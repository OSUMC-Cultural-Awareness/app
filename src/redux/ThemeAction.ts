import { Store } from ".";

export const updateTheme = (
  theme: Store["theme"]
): { type: string; payload: Store["theme"] } => ({
  type: "UPDATE_THEME",
  payload: theme,
});
