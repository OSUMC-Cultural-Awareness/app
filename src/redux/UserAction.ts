import { Store } from ".";

export const updateUser = (
  user: Store["user"]
): { type: string; payload: Store["user"] } => ({
  type: "UPDATE_USER",
  payload: user,
});

export const resetUser = (): {
  type: string;
  payload: Record<string, unknown>;
} => ({
  type: "RESET_USER",
  payload: {},
});
