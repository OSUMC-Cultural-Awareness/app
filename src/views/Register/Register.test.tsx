/**
 * @jest-environment jsdom
 */
import React from "react";
import "react-native";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";

import Register from ".";
import { USER_INITIAL_STATE, THEME_INITIAL_STATE } from "../../redux";
import { Routes } from "../../routes";

const RegisterTest = () => {
  const Stack = createStackNavigator<Routes>();

  const mockStore = configureStore([]);

  return (
    <Provider
      store={mockStore({
        user: USER_INITIAL_STATE,
        theme: THEME_INITIAL_STATE,
      })}
    >
      <NavigationContainer>
        <Stack.Navigator initialRouteName="Register">
          <Stack.Screen name="Register" component={Register} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
};

describe("Register", () => {
  beforeEach(() => {
    jest.useFakeTimers();
    jest.clearAllMocks();
  });

  test("Renders", () => {
    const rendered = renderer.create(<RegisterTest />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
