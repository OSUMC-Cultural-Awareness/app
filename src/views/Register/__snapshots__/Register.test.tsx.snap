// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Renders 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ForwardRef(NavigationContainer)>
    <StackNavigator
      initialRouteName="Register"
    >
      <Screen
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Register)",
            "type": [Function],
          }
        }
        name="Register"
      />
    </StackNavigator>
  </ForwardRef(NavigationContainer)>
</Provider>
`;
